### DISCUSSION TITLE
Log which query/mutation fields are used

### DISCUSSION THREAD

**Discussion by rudiedirkx** [webonyx/graphql-php#865] [Q&A]
Is there a way to detect which path the entire `GraphQL::executeQuery()` took? Right now I do a `Parser::parse(new Source($input['query']))` and then take the fields from the first definition, but that's not necessarily the used definition. It depends on `operationName`, which is sometimes optional etc etc.

Is there a way to let `GraphQL::executeQuery()` do what it does, and then afterwards ask what it did? Which operation, which main/root fields, how many fields in total, the max depth, etc etc?

**Comment by sebastienbarre** [comment]
I'd be interested in that as well, to collect some usage metrics.
Is there any middleware entrypoint that can be used?
Thanks

**Comment by le0daniel** [comment]
What I currently do is Proxy every resolve function to collect metrics...

```php
[
  "type" => Type:id(),
  "resolve" => new ProxyResolver(fn($data, $arguments, $context, $info) => $data['id']),
]
```
